<?xml version="1.0"?>
<launch>
    <!-- TR200 키보드 제어 전용 런치 파일 -->
    
    <!-- 키보드 제어 파라미터 (개발용 안전 설정) -->
    <arg name="linear_speed" default="0.05" />
    <arg name="angular_speed" default="0.1" />
    <arg name="command_timeout" default="2.0" />
    <arg name="publish_frequency" default="20.0" />
    
    <!-- 안전 설정 -->
    <arg name="enable_safety_limits" default="true" />
    <arg name="max_linear_speed" default="1.0" />
    <arg name="max_angular_speed" default="1.0" />
    
    <!-- Remote Control Node (키보드 입력 전용) -->
    <node 
        name="remote_control_node" 
        pkg="tr200_ros_control" 
        type="remote_control_node.py" 
        output="screen"
        respawn="false">
        
        <!-- 원격 제어 파라미터 -->
        <param name="linear_speed" value="$(arg linear_speed)" />
        <param name="angular_speed" value="$(arg angular_speed)" />
        <param name="command_timeout" value="$(arg command_timeout)" />
        <param name="publish_frequency" value="$(arg publish_frequency)" />
        
        <!-- 안전 설정 -->
        <param name="enable_safety_limits" value="$(arg enable_safety_limits)" />
        <param name="max_linear_speed" value="$(arg max_linear_speed)" />
        <param name="max_angular_speed" value="$(arg max_angular_speed)" />
    </node>
    
</launch>
